arangod/Transaction/Methods.cpp

```diff
@@ -3160,13 +3159,42 @@ Future<Result> Methods::replicateOperations(
   network::RequestOptions reqOpts;
   reqOpts.database = vocbase().name();
   reqOpts.param(StaticStrings::IsRestoreString, "true");
-  if (options.refillIndexCaches != RefillIndexCaches::kDefault) {
+
+  // index cache refilling...
+  if (options.refillIndexCaches == RefillIndexCaches::kDontRefill) {
+    // index cache refilling opt-out
+    reqOpts.param(StaticStrings::RefillIndexCachesString, "false");
+
+    TRI_IF_FAILURE("RefillIndexCacheOnFollowers::failIfFalse") {
+      THROW_ARANGO_EXCEPTION(TRI_ERROR_DEBUG);
+    }
+  } else {
     // this attribute can have 3 values: default, true and false. only
     // expose it when it is not set to "default"
```

Comment:
This statement is not true for your code.
Also I interested why

Reply:
Ok, internally it is an enum with 3 values.
On the API level, it is a boolean that can be either true (maps to `kRefill`) or false (maps to `kDontRefill). However, not specifying the URL parameter at all in a request leads to us not having any boolean value. This maps to `kDefault` (no value set, but use configured default value).